qml/MonitorItem.qml

```diff
@@ -25,6 +25,76 @@ Component
         UM.I18nCatalog { id: catalog; name:"mksplugin" }
         UM.I18nCatalog { id: cura_catalog; name: "cura"}
 
+        //camera view (experimental), begining
+        Rectangle {
+            Cura.NetworkMJPGImage {
+                id: cameraImage
+                
+                visible: OutputDevice.isCameraView
+                source: OutputDevice.cameraUrl //MJPG-streamer video stream URL
+                property real videoZoom: OutputDevice.cameraVideoZoom
+
+                property real videoWidth: Math.floor(imageWidth * videoZoom)
+                property real videoHeight: Math.floor(imageHeight * videoZoom)
+                property real modVideoWidth: Math.floor(videoWidth * (parent.height / videoHeight))
+                property real modVideoHeight: Math.floor(videoHeight * (parent.width / videoWidth))
+
+                width: {
+                    if (videoWidth < parent.width) {
+                        if (videoHeight < parent.height) {
+                            return videoWidth
+                        } else {
+                            return modVideoWidth
+                        }
+                    } else {
+                        if (modVideoWidth < parent.width) {
+                            return modVideoWidth
+                        } else {
+                            return parent.width
+                        }
+                    }
+                }
+                height: {
+                    if (videoHeight < parent.height) {
+                        if (videoWidth < parent.width) {
+                            return videoHeight
+                        } else {
+                            return modVideoHeight
+                        }
+                    } else {
+                        if (modVideoHeight < parent.height) {
+                            return modVideoHeight
+                        } else {
+                            return parent.height
+                        }
+                    }
+                }
+
+                anchors.horizontalCenter: parent.horizontalCenter
+                anchors.verticalCenter: parent.verticalCenter
+
+                Component.onCompleted: {
+                    if (visible) {
+                        start();
+                    }
+                }
+                onVisibleChanged: {
+                    if (visible) {
+                        start();
```

Comment:
This will be called only when user switched to the Monitor tab.
for example:
- manually for already connected printer
- after first click to the "connect" button from the plugin settings

When user tries to enable/disable settings, NetworkMJPGImage is unresponsive, because there is no signal emitter to update cameraUrl  and isCameraView values.
Test case:
- Open Monitor Tab with connected printer
  - Web camera stream is shown
- Open Plugin settings, go to the "advanced settings" tab
- Disable camera view feature
  - Monitor Tab not changed, camera stream is shown

As first implementation it's ok.  But you need to decide what to do: 
1. create a separate ticket for more realtime settings update
2. add signal emitter, connect it with MachineConfig and update NetworkMJPGImage state on change.

Reply:
Nice job!